### PR TITLE
new CESM single/column single/point functionality

### PR DESCRIPTION
**Pull request by mvertens**
### Description of changes
New single/column single/point functionality

### Specific notes
Implemented new nuopc/cmeps single column functionality. This PR only touches CESM cime_config and CESM driver code.
This PR no longer requires PR #160 .

1. In `config/cesm/config_files.xml` - single point domains are only used for mct/cpl7. For cmeps single point meshes are now generated on the fly and component domains files are no longer needed.
2. env_run.xml variables PTS_LAT, PTS_LON and PTS_DOMAINFILE are used to determine if there is a single point or single column run. 
  - If PTS_LAT and PTS_LON are not -999 and PTS_DOMAINFILE is UNSET, then you have a single point run and the exact values of PTS_LAT and PTS_LON are used. 
  - If PTS_LAT and PTS_LON are not -999 and PTS_DOMAINFILE is not UNSET, then then the cmeps driver will recognize the nearest neighbor values of PTS_LAT and PTS_LON in PTS_DOMAINFILE as the single column lat and lon to use.

This PR only effects nuopc/cmeps configurations and also needs an accompanying PR to CMEPS, CDEPS, CLM, CICE5 and CAM. A full documentation of the testing is done in the CMEPS PR.

Contributors other than yourself, if any:

CMEPS Issues Fixed: None

Are changes expected to change answers?
 - [x] bit for bit
 - [x] different at roundoff level (due to cdeps changes in the testing)
 - [ ] more substantial 

Any User Interface Changes (namelist or namelist defaults changes)?
 - [ ] Yes
 - [x] No

Testing performed if application target is CESM:(either UFS-S2S or CESM testing is required):
- [ ] (recommended) CIME_DRIVER=nuopc scripts_regression_tests.py
   - machines:
   - details (e.g. failed tests):
- [x] (recommended) CESM testlist_drv.xml
   - machines and compilers: cheyenne/intel
   - details (e.g. failed tests): 
   - in cdeps/datm changes to the calculation of nextsw_cday differ in the 16/17 decimal places with respect to the previous code
   - this results in answer changes of the new mar14 baseline with the mar12 baseline of the following
```
FAIL ERS_Vnuopc_Ld5.T31_g37_gl20.I1850Clm50SpG.cheyenne_intel.cism-test_coupling BASELINE mar12: DIFF
FAIL ERS_Vnuopc_Ld5.T62_g37.DTEST.cheyenne_intel BASELINE mar12: DIFF
FAIL ERS_Vnuopc_Ld5.f09_g17.I1850Clm50Sp.cheyenne_intel.clm-default BASELINE mar12: DIFF
FAIL ERS_Vnuopc_Ld5.f19_g17.I2000Clm51Bgc.cheyenne_intel.clm-default BASELINE mar12: DIFF
```
- [ ] (optional) CESM prealpha test
   - machines and compilers
   - details (e.g. failed tests):
- [ ] (other) please described in detail
   - machines and compilers
   - details (e.g. failed tests):

Testing performed if application target is UFS-coupled:
- [ ] (recommended) UFS-coupled testing
   - description:
   - details (e.g. failed tests):

Testing performed if application target is UFS-HAFS:
- [ ] (recommended) UFS-HAFS testing
   - description:
   - details (e.g. failed tests):

Hashes used for testing:
- [x] CESM:
  - repository to check out: https://github.com/ESCOMP/CESM.git
  - branch: nuopc_dev.scol
  - hash: 4fc946b
- [ ] UFS-coupled, then umbrella repostiory to check out and associated hash:
  - repository to check out:
  - branch:
  - hash:
- [ ] UFS-HAFS, then umbrella repostiory to check out and associated hash:
  - repository to check out:
  - branch:
  - hash:

### CODE REVIEW

File: drivers/cime/esm.F90 [hunk 1]
```diff
@@ -1101,6 +1074,281 @@ end subroutine esm_init_pelayout
 
   !================================================================================
 
+  subroutine esm_set_single_column_attributes(compname, gcomp, rc)
+
+    ! Generate a mesh for single column
+
+    use netcdf
```
Comment (jedwards4b) [inline drivers/cime/esm.F90:1081]: please add only clause here.

File: drivers/cime/esm.F90 [hunk 2]
```diff
@@ -1101,6 +1074,281 @@ end subroutine esm_init_pelayout
...
+
+          ! In this case the domain file is not a single point file - but normally a
+          ! global domain file where a nearest neighbor search will be done to find
+          ! the closest point in the domin file to scol_lon and scol_lat
+
+          status = nf90_open(single_column_lnd_domainfile, NF90_NOWRITE, ncid)
+          if (status /= nf90_noerr) then
+             call shr_sys_abort (trim(subname) //': Cannot open '//trim(single_column_lnd_domainfile))
+          endif
+          status = nf90_inq_dimid (ncid, 'ni', dimid)
```
Comment (jedwards4b) [inline drivers/cime/esm.F90:1191]: Should check status after each nf90 call

**Review by jedwards4b — CHANGES_REQUESTED** [review]

**Comment by jedwards4b** [comment]
@mvertens  I'll try to fix github action, it may involve several commits (but I got the cime one right the first time!)

**Review by jedwards4b — APPROVED** [review]

**Comment by jedwards4b** [comment]
@mvertens I added the action to tag the commit but it doesn't take affect until the action is on master, so after this branch is merged I will create another PR for the purpose of testing and implementing the tag feature.

**Review by DeniseWorthen — APPROVED** [review]
ufs-model does not use the files impacted by this PR.

File: drivers/cime/esm.F90 (hunk 2 above)
Comment (mvertens) [reply drivers/cime/esm.F90]: Done.

File: drivers/cime/esm.F90 (hunk 1 above)
Comment (mvertens) [reply drivers/cime/esm.F90]: Done.

**Comment by uturuncoglu** [comment]
@mvertens i have issue with syncing HAFS app. Once I solve it, I'll test both CDEPS and CMEPS under HAFS and let you know. Sorry for delay but the testing will be pretty strait forward in the near future since I have just finish the implementation of HAFS specific RTs in the UFS model side. I am still trying to test them on Orion but the response time is too slow at this point.

**Comment by uturuncoglu** [comment]
@mvertens since the mods in here isolated with driver and CIME interface I could approve this one and test CDEPS with CMEPS master.

**Review by uturuncoglu — APPROVED** [review]

**Pull request merged**